runtimes/peregrine/src/lib.rs

```diff
@@ -945,6 +945,36 @@ impl pallet_proxy::Config for Runtime {
 	type WeightInfo = weights::pallet_proxy::WeightInfo<Runtime>;
 }
 
+impl pallet_dip_provider::Config for Runtime {
+	type CommitOriginCheck = did::EnsureDidOrigin<DidIdentifier, AccountId>;
+	type CommitOrigin = did::DidRawOrigin<DidIdentifier, AccountId>;
+	type Identifier = DidIdentifier;
+	type IdentityCommitmentGenerator = runtime_common::dip::merkle::DidMerkleRootGenerator<Runtime>;
+	type IdentityProvider = runtime_common::dip::did::LinkedDidInfoProvider;
+	type ProviderHooks = pallet_dip_provider::NoopHooks;
+	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = ();
+}
+
+pub struct PreliminaryDipOriginFilter;
+
+impl frame_support::traits::Contains<RuntimeCall> for PreliminaryDipOriginFilter {
+	fn contains(_t: &RuntimeCall) -> bool {
+		true
+	}
+}
+
+impl pallet_dip_consumer::Config for Runtime {
+	type DipCallOriginFilter = PreliminaryDipOriginFilter;
+	type DispatchOriginCheck = EnsureSigned<AccountId>;
+	type Identifier = DidIdentifier;
+	type LocalIdentityInfo = u128;
+	type ProofVerifier = pallet_dip_consumer::traits::SuccessfulProofVerifier;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeOrigin = RuntimeOrigin;
+	type WeightInfo = ();
+}
+
```

Comment:
Please ignore this implementation. I will remove it once I get a green light. If this PR is approved, I will recalculate the default benchmarks in release mode and remove it.